build-tools-internal/src/main/groovy/elasticsearch.stable-api.gradle

```diff
@@ -0,0 +1,51 @@
+import org.elasticsearch.gradle.Version
+import org.elasticsearch.gradle.VersionProperties
+import org.elasticsearch.gradle.internal.BwcVersions
+import org.elasticsearch.gradle.internal.JarApiComparisonTask
+import org.elasticsearch.gradle.internal.info.BuildParams
+
+import java.util.stream.Collectors
+
+import static java.util.Arrays.stream
+
+configurations {
+  stableApi
+}
+
+BuildParams.bwcVersions.allIndexCompatible.findAll({ it.onOrAfter(Version.fromString("8.5.0")) && it != VersionProperties.elasticsearchVersion }).each { bwcVersion ->
+  def baseName = "v${bwcVersion}"
+
+  BwcVersions.UnreleasedVersionInfo unreleasedVersion = BuildParams.bwcVersions.unreleasedInfo(bwcVersion)
+  Configuration stableApiConfiguration = configurations.create("stableApi${baseName}") {
+    // TODO: Temporary workaround for https://github.com/elastic/elasticsearch/issues/73433
+    transitive = false
+  }
+  Object libDependency = null
+
+  if (unreleasedVersion) {
+    // For unreleased snapshot versions, build them from source
+    libDependency = files(project(unreleasedVersion.gradleProjectPath).tasks.named(ext.buildBwcName))
+    // TODO: is there a way to derive this from the project name instead?
+    // libDependency = files(project(unreleasedVersion.gradleProjectPath).tasks.named(buildBwcTaskName(project.name)))
```

Comment:
Certainly, we could copy `buildBwcTaskName` from `InternalDistributionBwcSetupPlugin` or just make it `public static` so we can reuse it here.